test/e2e/nsx_subnet_test.go

```diff
@@ -750,3 +793,547 @@ func fetchSubnetPortBySubnetPortUID(t *testing.T, subnetPortUID string) *model.V
+		if len(sharedSubnets) != expectedCount {
+			log.Info("Waiting for expected number of shared subnets",
+				"namespace", namespace,
+				"currentCount", len(sharedSubnets),
+				"expectedCount", expectedCount)
+			return false, nil
+		}
+
+		// If expectedCount is 0, we're done
+		if expectedCount == 0 {
+			return true, nil
+		}
+
+		// Verify name prefixes if provided and if we expect subnets
+		if len(namePrefixes) > 0 {
+			for _, subnet := range sharedSubnets {
+				hasExpectedPrefix := false
+				for _, prefix := range namePrefixes {
+					if strings.HasPrefix(subnet.Name, prefix) {
+						hasExpectedPrefix = true
+						break
+					}
+				}
+				if !hasExpectedPrefix {
+					log.Info("Shared subnet has unexpected name prefix",
+						"namespace", namespace,
+						"name", subnet.Name,
+						"expectedPrefixes", namePrefixes)
+					return false, nil
+				}
+			}
+		}
+
+		return true, nil
+	})
+
+	// Handle assertions based on expected count
+	if expectedCount == 0 {
+		require.NoError(t, err, "Shared subnet should be removed from %s", namespace)
+	} else {
+		require.NoError(t, err, "Failed to find %d shared subnet(s) in %s", expectedCount, namespace)
+		require.Equal(t, expectedCount, len(sharedSubnets),
+			"Expected %d shared subnet(s) in %s, got %d", expectedCount, namespace, len(sharedSubnets))
+
+		// Additional validation for single subnet case
+		if expectedCount == 1 && len(namePrefixes) > 0 {
+			require.NotNil(t, sharedSubnets[0], "Shared subnet in %s should not be nil", namespace)
+			require.True(t, strings.HasPrefix(sharedSubnets[0].Name, namePrefixes[0]),
+				"Shared subnet name should be prefixed with '%s', got: %s", namePrefixes[0], sharedSubnets[0].Name)
+		}
+	}
+
+	return sharedSubnets
+}
+
+// verifyNoSharedSubnet verifies that no shared subnet exists in the namespace
+func verifyNoSharedSubnet(t *testing.T, namespace string) {
+	verifySharedSubnets(t, namespace, 0)
+}
+
+// verifySharedSubnet verifies that a shared subnet with the expected name prefix exists in the namespace
+func verifySharedSubnet(t *testing.T, namespace string, namePrefix string) *v1alpha1.Subnet {
+	subnets := verifySharedSubnets(t, namespace, 1, namePrefix)
+	return subnets[0]
+}
+
+// verifyMultipleSharedSubnets verifies that multiple shared subnets with the expected name prefixes exist in the namespace
+func verifyMultipleSharedSubnets(t *testing.T, namespace string, expectedCount int, namePrefixes ...string) []*v1alpha1.Subnet {
+	return verifySharedSubnets(t, namespace, expectedCount, namePrefixes...)
+}
+
+// PrecreatedSharedSubnetBasic tests sharing a subnet with multiple namespaces
+func PrecreatedSharedSubnetBasic(t *testing.T) {
+	// The subnet is created directly in the target namespace using the NSX client REST API
+	// because the nsx-operator will append an underscore to the subnet name.
+	// When retrieving this name from nsx to create the CR again,
+	// the presence of an underscore causes the CR creation to fail
```

Comment:
Just to double confirm that do you mean testdata.nsxClient.SubnetsClient.Patch will append an underscore to the subnet name?